plugins/repository-hdfs/hadoop-common-patcher/src/main/java/org/elasticsearch/hdfs/patch/HdfsClassPatcher.java

```diff
@@ -41,18 +41,17 @@ public static void main(String[] args) throws Exception {
         try (JarFile jarFile = new JarFile(new File(jarPath))) {
             for (var patcher : patchers.entrySet()) {
                 JarEntry jarEntry = jarFile.getJarEntry(patcher.getKey());
-                if (jarEntry == null) {
-                    throw new IllegalArgumentException("path [" + patcher.getKey() + "] not found in [" + jarPath + "]");
```

Comment:
This had to go, as some classes are not present in both jars; it's unfortunate, but I think acceptable. If it's not, should we pass the classes to patch via the command line, so it is configurable in each gradle `JavaExec` task?

Reply:
we should pass the files to patch down, it should be part of the configuration of the task. we can keep how to actually patch the classes here, but the list of classes to patch be passed in.

However, why would the same class be present in two different jars? That's jarhell, which we should be rejecting already?

Reply:
I should have said both _projects_; the hdsf-fixture projects needs a subset of the patches, as it pulls in a different (smaller) set of dependencies.

Reply:
I'm still missing something. Why do dependencies matter? We should only be patching a given class once.